### ISSUE TITLE
False positive `@typescript-eslint/explicit-member-accessibility` for interface's getters/setters

### ISSUE THREAD

**Issue by elumixor** [eslint/eslint#14950]
**Tell us about your environment**

* **ESLint Version:** 7.32.0
* **Node Version:** 12.18.3
* **npm Version:** 6.14.6

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

`@typescript-eslint/parser`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

`.eslintrc.json`

```js
{
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaFeatures": {
            "modules": true,
            "jsx": true
        },
        "project": "./tsconfig.json",
        "warnOnUnsupportedTypeScriptVersion": false
    },
    "env": {
        "browser": true,
        "node": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:react/recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/recommended-requiring-type-checking",
        "prettier",
        "prettier/@typescript-eslint"
    ],
    "ignorePatterns": ["*.js"],
    "plugins": ["@typescript-eslint", "import", "react-hooks", "prefer-arrow"],
    "rules": {
        "@typescript-eslint/explicit-function-return-type": "off",
        "@typescript-eslint/naming-convention": [
            "warn",
            {
                "selector": "interface",
                "format": ["PascalCase"],
                "custom": {
                    "regex": "^I[A-Z]",
                    "match": true
                }
            },
            {
                "selector": ["variable", "function"],
                "format": ["camelCase"],
                "leadingUnderscore": "allow"
            },
            {
                "selector": "variable",
                "modifiers": ["destructured"],
                "format": null
            }
        ],
        "@typescript-eslint/member-ordering": ["warn"],
        "@typescript-eslint/no-empty-interface": "off",
        "@typescript-eslint/no-explicit-any": "off",
        "@typescript-eslint/no-misused-promises": [
            "error",
            {
                "checksVoidReturn": false
            }
        ],
        "@typescript-eslint/no-unused-vars": [
            "warn",
            {
                "argsIgnorePattern": "^props$",
                "ignoreRestSiblings": true
            }
        ],
        "@typescript-eslint/no-floating-promises": "warn",
        "@typescript-eslint/ban-types": "warn",
        "@typescript-eslint/restrict-template-expressions": "warn",
        "@typescript-eslint/restrict-plus-operands": "warn",
        "@typescript-eslint/no-unsafe-assignment": "off",
        "@typescript-eslint/no-unsafe-call": "off",
        "@typescript-eslint/no-unsafe-member-access": "off",
        "@typescript-eslint/no-unsafe-return": "off",
        "@typescript-eslint/no-empty-function": "warn",
        "@typescript-eslint/explicit-module-boundary-types": "off",
        "@typescript-eslint/no-use-before-define": "off",
        "@typescript-eslint/no-var-requires": "off",
        "@typescript-eslint/require-await": "off",
        "@typescript-eslint/explicit-member-accessibility": [
            "error",
            {
                "accessibility": "explicit",
                "overrides": {
                    "accessors": "explicit",
                    "constructors": "no-public",
                    "methods": "explicit",
                    "properties": "explicit",
                    "parameterProperties": "explicit"
                }
            }
        ],
        "id-blacklist": ["error", "Undefined", "any", "boolean", "number", "string"],
        "import/no-default-export": ["error"],
        "import/order": ["error"],
        "no-console": ["error"],
        "no-constant-condition": "off",
        "no-debugger": "off",
        "react-hooks/exhaustive-deps": "warn",
        "react-hooks/rules-of-hooks": "error",
        "react/display-name": "off",
        "react/react-in-jsx-scope": "off",
        "space-before-function-paren": [
            "error",
            {
                "anonymous": "never",
                "asyncArrow": "always",
                "named": "never"
            }
        ],
        "import/no-self-import": "error",
        "import/no-absolute-path": "error",
        "no-duplicate-imports": "off",
        "@typescript-eslint/no-duplicate-imports": ["error"]
    }
}
```

Packages

```js
"typescript": "4.3.5",
"eslint": "7.32.0",
"@typescript-eslint/eslint-plugin": "4.29.2",
"@typescript-eslint/parser": "4.23.0"
```
`tsconfig.json`
```js
{
  "compilerOptions": {
    "module": "esnext",
    "moduleResolution": "node",
    "target": "esnext",
    "downlevelIteration": true,
    "strict": true,
    "strictFunctionTypes": false,
    "types": ["node"],
    "sourceMap": true,
    "allowJs": true,
    "allowUnreachableCode": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "noUnusedLocals": true,
    "baseUrl": "./src/client",
    "skipLibCheck": true,
    "outDir": "./tsout",
    "allowSyntheticDefaultImports": true,
    "noImplicitOverride": true,
  },
  "exclude": ["node_modules", "dist"],
  "compileOnSave": false
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Declare interface with getters/setters (ts 4.3.5)

```typescript
export interface IFoo {
    get bar(): number;
    set bar(value: number);
}
```

**Expected**
No errors, as adding an accessibility modifier in an interface would be a typescript error.

**Actual**

I get these errors

```
@typescript-eslint/explicit-member-accessibility: Missing accessibility modifier on get property accessor bar.
@typescript-eslint/explicit-member-accessibility: Missing accessibility modifier on set property accessor bar.
```

Is there something I did wrong? Is this not yet supported by eslint?

**Comment by mdjermanovic** [comment]
Hi @elumixor, thanks for the issue!

We're not maintaining `@typescript-eslint/explicit-member-accessibility`, it's a rule from `@typescript-eslint/eslint-plugin`. Can you please open an issue in the plugin's repository?

https://github.com/typescript-eslint/typescript-eslint
